Development practices/File upload.md

```diff
@@ -0,0 +1,57 @@
+Our standard file upload setup includes:
+- using Shrine gem
+- direct S3 uploads
+- [S3](https://aws.amazon.com/s3/) as file storage for production and staging servers, and disk storage for development.
+
+# Shrine
+There are several file upload gems out there. We use [Shrine](https://github.com/janko-m/shrine) because of its flexibility and maintainability.
+It is a general-purpose file uploader gem. Since it has no direct dependency,
+__it can be used in both Rails and non-Rails projects.__
+
+Official docs can be found [here](https://shrinerb.com/)
+There is also [a demo application](https://github.com/erikdahlstrand/shrine-rails-example).
+
+## Plugins
+Shrine features a rich plugin system which makes it incredibly easy to change its behavior to suit your needs.
+
+Here are some of the plugins we use often:
+- [activerecord](https://shrinerb.com/docs/plugins/activerecord)
+- [bakgrounding](https://shrinerb.com/docs/plugins/backgrounding)
+- [cached_attachment_data](https://shrinerb.com/docs/plugins/cached_attachment_data) - for forms
+- [derivatives](https://shrinerb.com/docs/plugins/derivatives)
+- [determine_mime_type](https://shrinerb.com/docs/plugins/determine_mime_type)
+- [presign_endpoint](https://shrinerb.com/docs/plugins/presign_endpoint) - for direct S3 uploads
+- [upload_options](https://shrinerb.com/docs/plugins/upload_options)
+- [url_options](https://shrinerb.com/docs/plugins/url_options)
+- [validation](https://shrinerb.com/docs/validation)
+- [validation_helpers](https://shrinerb.com/docs/plugins/validation_helpers)
+
+## Security
+Files stored on S3 are private by default. This means file URLs will be signed and they will expire after some specified time.
+
+It's best to explicitly set the expiration time for **each uploader class** using [url_options](https://shrinerb.com/docs/plugins/url_options) plugin.
+
+Some files need to be public, i.e. albums' covers. In that case, set the `acl` to `public-read` via [upload_options](https://shrinerb.com/docs/plugins/upload_options) plugin for that uploader class.
```

Comment:
```suggestion
Some files need to be public, i.e. album covers. In that case, set the `acl` to `public-read` via the [upload_options](https://shrinerb.com/docs/plugins/upload_options) plugin for that uploader class.
```
The possessive form is not so necessary with inanimate stuff (and it's better if it's not there).